### ISSUE TITLE
`Could not compare snapshot`

### ISSUE THREAD

**Issue by vjpr** [wallabyjs/public#1428]
### Issue description or question

I am seeing `Could not compare snapshot` for tests which use to run fine.

I don't see it when I run tests from the command-line.

Version: `core v1.0.529`

### Code editor or IDE name and version

Visual Studio Code v1.?
WebStorm v?
**IntelliJ IDEA v?**
Atom v1.?
Visual Studio v?
Sublime Text v3 build ?

### OS name and version

Windows
**OSX**
Linux

**Comment by vjpr** [comment]
Filed a bug at ava too: https://github.com/avajs/ava/issues/1609

Seems that silence the real error which was `[ChecksumError: Checksum mismatch]`.

**Comment by vjpr** [comment]
My best guess is that the `.snap` files are being modified by wallaby somehow when they are copied to the cache dir.

**Comment by vjpr** [comment]
```

// wallabyCacheDir - instrumented

00000000: 4156 4120 536e 6170 7368 6f74 2076 310a  AVA Snapshot v1.
00000010: 0100 0eef bfbd efbf bdef bfbd 352f 62ef  ............5/b.
00000020: bfbd 1aef bfbd efbf bd15 0eef bfbd efbf  ................
00000030: bdef bfbd 1fef bfbd 0800 0000 0000 0013  ................
00000040: efbf bd62 6060 6062 efbf bdef bfbd efbf  ...b```b........
00000050: bd29 7eef bfbd efbf bd71 68ef bfbd 6bef  .)~......qh...k.
00000060: bfbd efbf bdef bfbd d58b 1819 40ef bfbd  ............@...
00000070: 1310 efbf bdef bfbd efbf bdef bfbd efbf  ................
00000080: bd5b 3bef bfbd efbf bd29 efbf bdef bfbd  .[;......)......
00000090: efbf bd51 1623 58ef bfbd 0923 4817 3b58  ...Q.#X....#H.;X
000000a0: 1523 3b23 efbf bd2b efbf bdef bfbd 06ef  .#;#...+........
000000b0: bfbd 6c20 efbf bd04 efbf bd25 40ef bfbd  ..l .......%@...
000000c0: 0fef bfbd 6f00 efbf bd08 23ef bfbd 1023  ....o.....#....#
000000d0: efbf bd7f 5256 6a72 0923 23ef bfbd 2928  ....RVjr.##...)(
000000e0: 28ef bfbd efbf bdef bfbd efbf bdef bfbd  (...............
000000f0: efbf bdef bfbd efbf bdef bfbd efbf bdef  ................
00000100: bfbd efbf bd0f efbf bdef bfbd 49ef bfbd  ............I...
00000110: efbf bdef bfbd efbf bdef bfbd 45ef bfbd  ............E...
00000120: efbf bd23 efbf bd28 35ef bfbd 393f 2515  ...#...(5...9?%.
00000130: 2cef bfbd efbf bdc8 8826 cf9a efbf bd5a  ,........&.....Z
00000140: efbf bd02 efbf bd64 7ac9 8c26 cb94 efbf  .......dz..&....
00000150: bd53 efbf bdef bfbd 022a efbf bd2d efbf  .S.......*...-..
00000160: bd2e 5309 efbf bdef bfbd efbf bd03 5def  ..S...........].
00000170: bfbd 0a2a efbf bd29 3cef bfbd 7c0c 0079  ...*...)<...|..y
00000180: efbf bdef bfbd 031e 0200 00              ...........

// original

00000000: 4156 4120 536e 6170 7368 6f74 2076 310a  AVA Snapshot v1.
00000010: 0100 0ef9 8990 352f 62cf 1acf e115 0ee8  ......5/b.......
00000020: c5cf 1f8b 0800 0000 0000 0013 b362 6060  .............b``
00000030: 6062 90ef ed97 297e b6e0 7168 886b 9ef3  `b....)~..qh.k..
00000040: d6d5 8b18 1940 e013 10f7 abb2 e9b9 dc5b  .....@.........[
00000050: 3ba9 b029 cdcf e99c 5116 2358 f409 2348  ;..)....Q.#X..#H
00000060: 173b 5815 233b 23bb 2b90 f606 e26c 20ee  .;X.#;#.+....l .
00000070: 04e2 2540 bc0f 886f 00b1 0823 bb10 239b  ..%@...o...#..#.
00000080: 7f52 566a 7209 2323 9c29 2828 c8f4 bf9e  .RVjr.##.)((....
00000090: 8191 8781 918d 8191 0fa8 8c49 9491 9981  ...........I....
000000a0: 9145 8891 23b1 2835 d139 3f25 152c c0cc  .E..#.(5.9?%.,..
000000b0: c888 26cf 9a92 5ae6 9902 9164 7ac9 8c26  ..&...Z....dz..&
000000c0: cb94 8953 8ab1 022a a32d 842e 5309 95f9  ...S...*.-..S...
000000d0: fa03 5da6 0a2a c329 3cf2 7c0c 0079 a6f1  ..]..*.)<.|..y..
000000e0: 031e 0200 00
```

**Comment by vjpr** [comment]
@ArtemGovorov 

So seems like your are reading all files in `utf8` format.

```
const file = require('fs').readFileSync('index.test.js.snap', 'utf8')
const out = Buffer.from(file).toString('hex')

// `out` is now the same as the wallabyCacheDir now - which should not be.
```

**Comment by vjpr** [comment]
Seems to be related to: https://github.com/wallabyjs/public/issues/947

**Comment by vjpr** [comment]
Oh man I was so close to finding the `binary` option in the docs, I just didn't scroll down enough!

What is the difference when copying whether its binary or text?

The issue seems to be that it failed to be detected as binary because there is ascii at the start of the binary file - or its just not a common format.

**Comment by ArtemGovorov** [comment]
Wallaby needs to identify binary files (to correctly process all files), and Wallaby for IntelliJ platform is using IntelliJ API to find out if a file is binary or not. The response from the API (whether the file is binary or not) depends on the IntelliJ platform guess + your file association settings. Looks like in your case IDEA doesn't think that your file is binary for some reason.

> Seems to be related to: https://github.com/wallabyjs/public/issues/947 https://github.com/wallabyjs/public/issues/1177 https://github.com/wallabyjs/public/issues/614 https://github.com/wallabyjs/public/issues/521 https://github.com/wallabyjs/public/issues/1394

These are all VS Code/Atom related issues, related to a different way of finding out whether a file is binary or not in those editors.

To tackle the scenario of when VS Code/Atom (and node module whitelisting some known binary extensions) fails to correctly identify whether a file is binary or not, we have introduced the `binary` flag. 

> The `binary` option is not working for me :(

The flag so far had only been used by VS Code/Atom implementation, and IntelliJ platform integration didn't really need it as its API provided a reliable way to identify binary files. However, it looks like you've hit some limitation of the API (or maybe your IDE settings are involved somehow). 

Anyway, we have just added the `binary` flag support to IntelliJ integration as well. Try upgrading your plugin to `1.0.101`, add the `binary` flag to your `*.snap` file pattern/clear cache and try starting Wallaby.

**Comment by vjpr** [comment]
> Try upgrading your plugin to 1.0.101, add the binary flag to your *.snap file pattern/clear cache and try starting Wallaby.

Didn't work.

**Comment by ArtemGovorov** [comment]
> At least it would be good to add `.snap` as default binary files.

Jest's snapshots are normally text files, and the `.snap` extension is used by Jest by default for all of their snapshots, so I don't think adding `.snap` extension as a binary one for everyone is a good idea.

> Didn't actually work.

It's my fault as it turned out that not all of the commits related to the `binary` flag support for IntelliJ platform landed in v1.0.101. We have released v1.0.102 version of the plugin that should now include all of the changes to make it work (with the `binary` flag and without requiring any changes in your IDE settings). Try upgrading, it should work for you now.
